test/index.test.ts

```diff
@@ -49,6 +49,9 @@ describe('Autoapproval bot', () => {
 
     // Receive a webhook event
     await probot.receive({ name: 'pull_request', payload })
+
+    await new Promise(process.nextTick) // Don't assert until all async processing finishes
```

Comment:
This particular bit of waiting is necessary because even though our GitHub App is async and we are awaiting its call just above, that function fires off additional async calls that are _not_ awaited (i.e. the interactions with the GitHub API) and this assertion ends up being evaluated prior to those processes finishing.

Reply:
I don't really understand how previously the assertions work then 🤔

Reply:
After reading [the `nock` documentation](https://github.com/nock/nock#expectations), I don't think there were any assertions at all 😬 Or at least, not sufficient ones.

Calling `nock(...).blah(...)` will create a mock and register a 'scope' for a particular domain, and it will raise an error if an unmocked request is made within that scope (i.e. to that same domain). But it doesn't complain if the request being mocked isn't actually made: the way to do that is by calling [`nock.isDone()`](https://github.com/nock/nock#isdone), so I added those calls to the tests.

It turns out that all the existing tests were fine without my changes, but when I added these assertions, some of them started failing on my feature branch; that's how I knew I needed to update the test fixtures 😅

Reply:
Nice findings! Thanks for sharing 👍